src/cpp/middleware/fast/FastEntities.cpp

```diff
@@ -243,30 +316,32 @@ bool FastDataReader::match_from_xml(const std::string& xml) const
     return rv;
 }
 
-bool FastDataReader::set_on_new_data_cb(OnNewData on_new_data_cb)
-{
-    on_new_data_cb_ = on_new_data_cb;
-    return true;
-}
-
-bool FastDataReader::unset_on_new_data_cb()
-{
-    on_new_data_cb_ = nullptr;
-    return true;
-}
-
-bool FastDataReader::read(std::vector<uint8_t>* data)
+bool FastDataReader::read(std::vector<uint8_t>* data, uint32_t timeout)
 {
+    auto now = std::chrono::steady_clock::now();
```

Comment:
Can this line can be put just before the `wait_until` (inside of `else` branch) or the line is it put before the lock explicity?

Reply:
No, imagine that takes the `locks` takes you more than the timeout.
In fact, I have to modify the timeout pass to the `wait_until`.